### ISSUE TITLE
no-unsafe-any fails for `implements Namespace.Interface`

### ISSUE THREAD

**Issue by MayaWolf** [palantir/tslint#3255]
### Bug Report

- __TSLint version__: 5.7.0
- __TypeScript version__: 2.5.2
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
namespace TestNS {
    export interface ITest {}
}
class Test implements TestNS.ITest {}
```

with `tslint.json` configuration:

```json
"no-unsafe-any": true
```

#### Actual behavior
`(6,23): Unsafe use of expression of type 'any'.`

#### Expected behavior
No warning.

#### Reason
[This happens here.](https://github.com/palantir/tslint/blob/master/src/rules/noUnsafeAnyRule.ts#L151) For some reason, `checker.getTypeAtLocation` doesn't work for namespaces in `implements` and instead just returns an unknown `any` type.

However, unless I'm missing something, any `SyntaxKind.HeritageClause` could be excluded from `any` checking (i.e. added around [here](https://github.com/palantir/tslint/blob/master/src/rules/noUnsafeAnyRule.ts#L77)), since the compiler should handle those errors anyway. I have tested this and it seems to fix the problem.

**Comment by vladeck** [comment]
I have this issue as well with **TSLint 5.8.0**.

**Comment by ajafff** [comment]
> However, unless I'm missing something, any SyntaxKind.HeritageClause could be excluded from any checking (i.e. added around here), since the compiler should handle those errors anyway.

That's not quite right. It's not safe for classes to extend `any`, because the call to the super constructor might fail due to the lack of type checking.

> For some reason, checker.getTypeAtLocation doesn't work for namespaces in implements and instead just returns an unknown any type.

That's probably because `TestNS.ITest` has no type, it **is** a type. There's a separate API for that: `checker.getDeclaredType...`

**Comment by ajafff** [comment]
Fix is up in #3487
Thanks for the bump on this issue.

**Comment by geoffadams** [comment]
We're seeing this issue when using `pg-promise` with `no-unsafe-any` enabled. At least, I believe so.

I notice there is a pull request open for this with some problems preventing it from being merged. Just wanted to help bump this issue by demonstrating another error scenario.

Code being linted:
```
import * as pgPromise from 'pg-promise'

const pg = pgPromise()
const db = pg('postgres://user:pass@localhost/database')
```

Type definitions can be found here: https://github.com/vitaly-t/pg-promise/blob/a6429cda84b7d7a9d43d0fa4e6d9c884ffdb9cfc/typescript/pg-promise.d.ts

Error:
```
ERROR: .../src/sample.ts[3, 12]: Unsafe use of expression of type 'any'.
ERROR: .../src/sample.ts[4, 12]: Unsafe use of expression of type 'any'.
```
